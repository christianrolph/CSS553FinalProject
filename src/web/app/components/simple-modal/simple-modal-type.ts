/**
 * Represents the type of  the modal to be displayed.
 */
export enum SimpleModalType {
  /**
   * Warning modal.
   */
  WARNING,
  /**
   * Danger modal.
   */
  DANGER,
  /**
   * Info modal.
   */
  INFO,
  /**
   * Neutral modal.
   */
  NEUTRAL,
  /**
   * Load modal.
   */
  LOAD,
}
