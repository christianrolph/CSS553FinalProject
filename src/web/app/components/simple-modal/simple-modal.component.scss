.margin-left-5px {
  margin-left: 5px;
}

.progress-bar-container {
  margin-top: 10px;
  margin-bottom: 2px;
}

.transparent-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  visibility: false;
  z-index: 0;
}

.display-none {
  display: none;
}

.z-index-1 {
  z-index: 1;
}
