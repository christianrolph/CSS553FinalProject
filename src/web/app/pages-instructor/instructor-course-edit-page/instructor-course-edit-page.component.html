<tm-loading-retry [shouldShowRetry]="hasCourseLoadingFailed" [message]="'Failed to load course'" (retryEvent)="loadCourseInfo()">
  <div *tmIsLoading="isCourseLoading" class="card bg-light top-padded text-center">
    <div class="card-header bg-primary text-white">
      <strong>Course</strong>
      <div class="card-header-btn-toolbar">
        <button id="btn-edit-course" class="btn btn-primary btn-sm"
                [disabled]="!currInstructorCoursePrivilege.canModifyCourse"
                *ngIf="!isEditingCourse" (click)="this.isEditingCourse = true">
          <i class="fa fa-pen" aria-hidden="true"></i>
          Edit
        </button>
        <button id="btn-cancel-course" *ngIf="isEditingCourse" class="btn btn-primary btn-sm"
                [disabled]="isSavingCourseEdit"
                (click)="cancelEditingCourse()">
          <i class="fa fa-times" aria-hidden="true"></i>
          Cancel
        </button>
        <button id="btn-delete-course" class="btn btn-primary btn-sm" ngbTooltip="Delete the course and its corresponding students and sessions"
                [disabled]="!currInstructorCoursePrivilege.canModifyCourse || isSavingCourseEdit" (click)="deleteCourse()">
          <i class="fa fa-trash" aria-hidden="true"></i>
          Delete
        </button>
      </div>
    </div>
    <div class="card-body fill-plain">
      <form #courseForm="ngForm" (ngSubmit)="onSaveCourse()">
        <div class="form-row form-group">
          <label class="col-sm-3 font-weight-bold text-md-right">Course ID:</label>
          <div class="col-sm-9">
            <input id="course-id" class="col-xs-12 form-control" type="text" tabindex="1"
                   #courseId="ngModel" [(ngModel)]="course.courseId" name="courseId"
                   [maxlength]="FormValidator.COURSE_ID_MAX_LENGTH" required disabled>
          </div>
        </div>
        <div class="form-row form-group">
          <label class="col-sm-3 font-weight-bold text-md-right">Course Name:</label>
          <div class="col-sm-9">
            <input id = "course-name" class="form-control"
                   type="text" tabindex="2" [disabled]="!isEditingCourse" name="courseName"
                   [(ngModel)]="course.courseName" #courseName="ngModel"
                   [maxlength]="FormValidator.COURSE_NAME_MAX_LENGTH" required>
            <div [hidden]="courseName.valid || (courseName.pristine && courseName.untouched)"
                 class="invalid-field">
              <i class="fa fa-exclamation-circle" aria-hidden="true"></i>
              The field Course Name should not be empty.
            </div>
          </div>
        </div>
        <div class="form-row form-group">
          <label class="col-xs-12 col-sm-3 font-weight-bold text-md-right"><span class="ngb-tooltip-class" ngbTooltip="You should not need to change this as it is auto-detected based on your
                device settings. TEAMMATES automatically adjusts to match the current time offset in your area, including clock changes due to daylight saving time.">
            Time Zone:</span></label>
          <div class="col-xs-12 col-sm-9">
            <div class="input-group">
              <select id="time-zone" class="form-control"
                [(ngModel)]="course.timeZone" #timeZone="ngModel" [disabled]="!isEditingCourse" required name="timeZone">
                <option *ngFor="let timezone of timezones" [value]="timezone.id">{{ timezone.id }} ({{ timezone.offset }})</option>
              </select>
              <span class="input-group-btn">
                <button class="btn btn-primary" type=button [disabled]="!isEditingCourse || isSavingCourseEdit" (click)="detectTimezone()">
                  Auto-Detect
                </button>
               </span>
            </div>
          </div>
        </div>
        <div class="form-group m-0">
          <div class=" col-sm-12 text-center">
            <button id="btn-save-course" class="btn btn-success" type="submit" [disabled]="form.invalid || isSavingCourseEdit" *ngIf="isEditingCourse">
              <tm-ajax-loading *ngIf="isSavingCourseEdit"></tm-ajax-loading>
              Save Changes
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
  <div class="right top-padded">
    <a class="small" tmRouterLink="/web/instructor/help" [queryParams]="{questionId: CoursesSectionQuestions.COURSE_INSTRUCTOR_ACCESS, section: Sections.courses}" rel="noopener noreferrer" target="_blank">
      <i class="fa fa-info-circle" aria-hidden="true"></i>
      More about configuring access permissions
    </a>
  </div>
</tm-loading-retry>

<tm-loading-retry [shouldShowRetry]="hasInstructorsLoadingFailed" [message]="'Failed to load instructors of course'" (retryEvent)="loadCourseInstructors()">
  <div *tmIsLoading="isInstructorsLoading">
    <ng-container *ngFor="let panelDetail of instructorDetailPanels; let i = index">
      <tm-instructor-edit-panel
          [editMode]="EditMode.EDIT" [(instructor)]="panelDetail.editPanel" [instructorIndex]="i"
          [currInstructorCoursePrivilege]="currInstructorCoursePrivilege"
          (viewRolePrivilegeModel)="viewRolePrivilegeModel($event)"
          (deleteInstructor)="deleteInstructor(i)"
          (saveInstructor)="saveInstructor(i)"
          (cancelEditing)="cancelEditingInstructor(i)"
          (sendRemindJoinEmail)="resendReminderEmail(i)"
          [allSections]="this.allSections" [allSessions]="this.allSessions">
      </tm-instructor-edit-panel>
      <br/>
    </ng-container>
  </div>
</tm-loading-retry>
<div class="text-center" *ngIf="!isAddingNewInstructor">
  <button id="btn-add-instructor" class="btn btn-success" tabindex="6" type="button" (click)="isAddingNewInstructor = true">
    Add New Instructor
  </button>
</div>
<div *ngIf="isAddingNewInstructor" @collapseAnim>
  <tm-instructor-edit-panel [editMode]="EditMode.ADD" [(instructor)]="newInstructorPanel" [instructorIndex]="instructorDetailPanels.length"
                            [isSavingNewInstructor]="isSavingNewInstructor"
                            [currInstructorCoursePrivilege]="currInstructorCoursePrivilege"
                            (viewRolePrivilegeModel)="viewRolePrivilegeModel($event)"
                            (saveInstructor)="addNewInstructor()" (cancelEditing)="isAddingNewInstructor = false;"
                            [allSections]="this.allSections" [allSessions]="this.allSessions"></tm-instructor-edit-panel>
</div>
