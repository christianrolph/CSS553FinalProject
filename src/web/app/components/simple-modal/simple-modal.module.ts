import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ProgressBarModule } from '../../components/progress-bar/progress-bar.module';
import { SimpleModalComponent } from './simple-modal.component';

/**
 * Module for the modal component
 */
@NgModule({
  declarations: [SimpleModalComponent],
  exports: [SimpleModalComponent],
  imports: [
    CommonModule,
    ProgressBarModule,
  ],
  entryComponents: [SimpleModalComponent],
})

export class SimpleModalModule { }
