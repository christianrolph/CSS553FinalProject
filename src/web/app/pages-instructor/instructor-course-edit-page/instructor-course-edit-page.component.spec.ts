import { HttpClientTestingModule } from '@angular/common/http/testing';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { FormsModule } from '@angular/forms';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { of } from 'rxjs';
import { CourseService } from '../../../services/course.service';
import { InstructorService } from '../../../services/instructor.service';
import { Course, Instructor, InstructorPermissionRole, JoinState } from '../../../types/api-output';
import { InstructorCreateRequest } from '../../../types/api-request';
import { AjaxLoadingModule } from '../../components/ajax-loading/ajax-loading.module';
import { LoadingRetryModule } from '../../components/loading-retry/loading-retry.module';
import { LoadingSpinnerModule } from '../../components/loading-spinner/loading-spinner.module';
import { SimpleModalModule } from '../../components/simple-modal/simple-modal.module';
import { TeammatesCommonModule } from '../../components/teammates-common/teammates-common.module';
import { TeammatesRouterModule } from '../../components/teammates-router/teammates-router.module';
import { CustomPrivilegeSettingPanelComponent } from './custom-privilege-setting-panel/custom-privilege-setting-panel.component';
import { InstructorCourseEditPageComponent } from './instructor-course-edit-page.component';
import {
    InstructorEditPanel,
    InstructorEditPanelComponent,
} from './instructor-edit-panel/instructor-edit-panel.component';
import { ViewRolePrivilegesModalComponent } from './view-role-privileges-modal/view-role-privileges-modal.component';

const testCourse: Course = {
  courseId: 'exampleId',
  courseName: 'Example Course',
  timeZone: 'UTC (UTC)',
  creationTimestamp: 0,
  deletionTimestamp: 1000,
};

const testInstructor1: Instructor = {
  courseId: 'exampleId',
  email: 'instructor1@gmail.com',
  joinState: JoinState.JOINED,
  name: 'Instructor 1',
};

const testInstructor2: Instructor = {
  courseId: 'exampleId',
  email: 'instructor2@gmail.com',
  joinState: JoinState.NOT_JOINED,
  name: 'Instructor 2',
};

const testInstructor3: Instructor = {
  courseId: 'exampleId',
  email: 'instructor3@gmail.com',
  joinState: JoinState.NOT_JOINED,
  name: 'Instructor 3',
};

const emptyInstructorPanel: InstructorEditPanel = {
  googleId: '',
  courseId: '',
  email: '',
  isDisplayedToStudents: true,
  displayedToStudentsAs: '',
  name: '',
  role: InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER,
  joinState: JoinState.NOT_JOINED,

  permission: {
    privilege: {
      canModifyCourse: true,
      canModifySession: true,
      canModifyStudent: true,
      canModifyInstructor: true,
      canViewStudentInSections: true,
      canModifySessionCommentsInSections: true,
      canViewSessionInSections: true,
      canSubmitSessionInSections: true,
    },
    sectionLevel: [],
  },

  isEditing: true,
  isSavingInstructorEdit: false,
};

describe('InstructorCourseEditPageComponent', () => {
  let component: InstructorCourseEditPageComponent;
  let fixture: ComponentFixture<InstructorCourseEditPageComponent>;
  let courseService: CourseService;
  let instructorService: InstructorService;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        InstructorCourseEditPageComponent,
        InstructorEditPanelComponent,
        ViewRolePrivilegesModalComponent,
        CustomPrivilegeSettingPanelComponent,
      ],
      imports: [
        NgbModule,
        FormsModule,
        AjaxLoadingModule,
        TeammatesCommonModule,
        RouterTestingModule,
        HttpClientTestingModule,
        LoadingSpinnerModule,
        LoadingRetryModule,
        TeammatesRouterModule,
        SimpleModalModule,
        BrowserAnimationsModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InstructorCourseEditPageComponent);
    component = fixture.componentInstance;
    courseService = TestBed.inject(CourseService);
    instructorService = TestBed.inject(InstructorService);
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should load correct course details for given API output', () => {
    spyOn(courseService, 'getCourseAsInstructor').and.returnValue(of(testCourse));

    component.loadCourseInfo();

    expect(component.course.courseId).toBe('exampleId');
    expect(component.course.courseName).toBe('Example Course');
    expect(component.course.timeZone).toBe('UTC (UTC)');
    expect(component.course.creationTimestamp).toBe(0);
    expect(component.course.deletionTimestamp).toBe(1000);
    expect(component.hasCourseLoadingFailed).toBeFalsy();
  });

  it('should not change course details if CANCEL is requested', () => {
    component.course = testCourse;
    component.isCourseLoading = false;
    component.originalCourse = Object.assign({}, component.course);
    fixture.detectChanges();

    component.isEditingCourse = true;
    component.course.courseName = 'Example Course Changed';
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement.querySelector('#btn-cancel-course');
    button.click();

    expect(component.isEditingCourse).toBeFalsy();
    expect(component.course.courseName).toBe('Example Course');
  });

  it('should update course details if SAVE is requested', () => {
    component.course = testCourse;
    component.isCourseLoading = false;
    fixture.detectChanges();

    component.isEditingCourse = true;
    component.course.courseName = 'Example Course Changed';
    fixture.detectChanges();

    spyOn(courseService, 'updateCourse').and.returnValue(of({
      courseId: 'exampleId',
      courseName: 'Example Course Changed',
      timeZone: 'UTC (UTC)',
      creationTimestamp: 0,
      deletionTimestamp: 1000,
    }));

    const button: any = fixture.debugElement.nativeElement.querySelector('#btn-save-course');
    button.click();

    expect(component.isEditingCourse).toBeFalsy();
    expect(component.course.courseName).toBe('Example Course Changed');
  });

  it('should load correct instructors details for given API output', () => {
    spyOn(instructorService, 'loadInstructors').and.returnValue(of({
      instructors: [testInstructor1, testInstructor2],
    }));

    component.loadCourseInstructors();

    expect(component.instructorDetailPanels[0].originalInstructor).toEqual(testInstructor1);
    expect(component.instructorDetailPanels[1].originalInstructor).toEqual(testInstructor2);
    expect(component.isInstructorsLoading).toBeFalsy();
  });

  it('should not add instructor if CANCEL is requested', () => {
    component.course = testCourse;
    component.isCourseLoading = false;
    component.instructorDetailPanels = [
      {
        originalInstructor: Object.assign({}, testInstructor1),
        originalPanel: component.getInstructorEditPanelModel(testInstructor1),
        editPanel: component.getInstructorEditPanelModel(testInstructor1),
      },
      {
        originalInstructor: Object.assign({}, testInstructor2),
        originalPanel: component.getInstructorEditPanelModel(testInstructor2),
        editPanel: component.getInstructorEditPanelModel(testInstructor2),
      },
    ];
    component.isAddingNewInstructor = true;
    component.newInstructorPanel = component.getInstructorEditPanelModel(testInstructor3);
    component.newInstructorPanel.isEditing = true;
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement
        .querySelector(`#btn-cancel-instructor-${ component.instructorDetailPanels.length + 1 }`);
    button.click();

    expect(component.isAddingNewInstructor).toBeFalsy();
  });

  it('should add instructor details', () => {
    spyOn(instructorService, 'createInstructor').and
      .callFake(({ courseId, requestBody }: { courseId: string, requestBody: InstructorCreateRequest }) => of({
        courseId,
        email: requestBody.email,
        joinState: JoinState.NOT_JOINED,
        name: requestBody.name,
      }));

    component.course = testCourse;
    component.courseId = testCourse.courseId;
    component.isCourseLoading = false;
    component.instructorDetailPanels = [
      {
        originalInstructor: Object.assign({}, testInstructor1),
        originalPanel: component.getInstructorEditPanelModel(testInstructor1),
        editPanel: component.getInstructorEditPanelModel(testInstructor1),
      },
      {
        originalInstructor: Object.assign({}, testInstructor2),
        originalPanel: component.getInstructorEditPanelModel(testInstructor2),
        editPanel: component.getInstructorEditPanelModel(testInstructor2),
      },
    ];
    component.isAddingNewInstructor = true;
    component.newInstructorPanel = component.getInstructorEditPanelModel(testInstructor3);
    component.newInstructorPanel.isEditing = true;
    fixture.detectChanges();

    const button: any = fixture.debugElement.nativeElement
        .querySelector(`#btn-save-instructor-${ component.instructorDetailPanels.length + 1 }`);
    button.click();

    expect(component.isAddingNewInstructor).toBeFalsy();
    expect(component.isSavingNewInstructor).toBeFalsy();
    expect(component.instructorDetailPanels.length).toBe(3);
    expect(component.instructorDetailPanels[2].originalInstructor).toEqual(testInstructor3);
    expect(component.newInstructorPanel).toEqual(emptyInstructorPanel);
  });

  it('should re-order if instructor is deleted', () => {
    spyOn(instructorService, 'deleteInstructor').and.returnValue(of({}));

    component.course = testCourse;
    component.isCourseLoading = false;
    component.instructorDetailPanels = [
      {
        originalInstructor: Object.assign({}, testInstructor1),
        originalPanel: component.getInstructorEditPanelModel(testInstructor1),
        editPanel: component.getInstructorEditPanelModel(testInstructor1),
      },
      {
        originalInstructor: Object.assign({}, testInstructor2),
        originalPanel: component.getInstructorEditPanelModel(testInstructor2),
        editPanel: component.getInstructorEditPanelModel(testInstructor2),
      },
    ];

    component.deleteInstructor(0);
    fixture.detectChanges();

    // using document instead of fixture as modal gets added into the dom outside the viewRef
    const button: any = document.getElementsByClassName('modal-btn-ok').item(0);
    button.click();
    fixture.detectChanges();

    expect(component.instructorDetailPanels.length).toBe(1);
    expect(component.instructorDetailPanels[0].originalInstructor).toEqual(testInstructor2);
  });

  it('should re-send reminder email for new instructors', () => {
    const mockReminderFunction: jest.MockedFunction<any> = jest.fn((_: string, email: string) => of({
      message: `An email has been sent to ${email}`,
    }));
    spyOn(courseService, 'remindInstructorForJoin').and.callFake(mockReminderFunction);

    component.course = testCourse;
    component.isCourseLoading = false;
    component.instructorDetailPanels = [
      {
        originalInstructor: Object.assign({}, testInstructor1),
        originalPanel: component.getInstructorEditPanelModel(testInstructor1),
        editPanel: component.getInstructorEditPanelModel(testInstructor1),
      },
      {
        originalInstructor: Object.assign({}, testInstructor2),
        originalPanel: component.getInstructorEditPanelModel(testInstructor2),
        editPanel: component.getInstructorEditPanelModel(testInstructor2),
      },
    ];
    fixture.detectChanges();

    let button: any = fixture.debugElement.nativeElement
        .querySelector(`#btn-resend-invite-${ component.instructorDetailPanels.length }`);
    button.click();

    // using document instead of fixture as modal gets added into the dom outside the viewRef
    button = document.getElementsByClassName('modal-btn-ok').item(0);
    button.click();

    expect(mockReminderFunction).toBeCalledWith(testCourse.courseId, testInstructor2.email);
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with course details', () => {
    component.course = testCourse;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should snap when editing course details', () => {
    component.isEditingCourse = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should snap with some instructor details', () => {
    const instructor: Instructor = {
      name: 'Instructor A',
      email: 'instructora@example.com',
      courseId: component.courseId,
      joinState: JoinState.JOINED,
    };

    component.instructorDetailPanels = [
      {
        originalInstructor: instructor,
        originalPanel: component.getInstructorEditPanelModel(instructor),
        editPanel: component.getInstructorEditPanelModel(instructor),
      },
    ];

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
