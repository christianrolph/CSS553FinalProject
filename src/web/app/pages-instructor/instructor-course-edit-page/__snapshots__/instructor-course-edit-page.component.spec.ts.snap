// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorCourseEditPageComponent should snap when editing course details 1`] = `
<tm-instructor-course-edit-page
  CoursesSectionQuestions={[Function Object]}
  EditMode={[Function Object]}
  FormValidator={[Function Object]}
  Sections={[Function Object]}
  allSections={[Function Array]}
  allSessions={[Function Array]}
  authService={[Function AuthService]}
  course={[Function Object]}
  courseId="undefined"
  courseService={[Function CourseService]}
  currInstructorCoursePrivilege={[Function Object]}
  currInstructorGoogleId=""
  feedbackSessionsService={[Function FeedbackSessionsService]}
  form="undefined"
  hasCourseLoadingFailed="false"
  hasInstructorsLoadingFailed="false"
  instructorDetailPanels={[Function Array]}
  instructorService={[Function InstructorService]}
  isAddingNewInstructor="false"
  isCourseLoading={[Function Boolean]}
  isEditingCourse={[Function Boolean]}
  isInstructorsLoading="false"
  isSavingCourseEdit="false"
  isSavingNewInstructor="false"
  navigationService={[Function NavigationService]}
  newInstructorPanel={[Function Object]}
  ngbModal={[Function NgbModal]}
  originalCourse={[Function Object]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
  timezones={[Function Array]}
>
  <tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    <div
      class="right top-padded"
    >
      <a
        class="small"
        href="/web/instructor/help?questionId=course-instructor-access&section=courses"
        rel="noopener noreferrer"
        target="_blank"
        tmrouterlink="/web/instructor/help"
      >
        <i
          aria-hidden="true"
          class="fa fa-info-circle"
        />
         More about configuring access permissions 
      </a>
    </div>
    
    
  </tm-loading-retry><tm-loading-retry>
    
    
    <div>
      
    </div>
    
    
  </tm-loading-retry><div
    class="text-center"
  >
    <button
      class="btn btn-success"
      id="btn-add-instructor"
      tabindex="6"
      type="button"
    >
       Add New Instructor 
    </button>
  </div>
</tm-instructor-course-edit-page>
`;

exports[`InstructorCourseEditPageComponent should snap with course details 1`] = `
<tm-instructor-course-edit-page
  CoursesSectionQuestions={[Function Object]}
  EditMode={[Function Object]}
  FormValidator={[Function Object]}
  Sections={[Function Object]}
  allSections={[Function Array]}
  allSessions={[Function Array]}
  authService={[Function AuthService]}
  course={[Function Object]}
  courseId="undefined"
  courseService={[Function CourseService]}
  currInstructorCoursePrivilege={[Function Object]}
  currInstructorGoogleId=""
  feedbackSessionsService={[Function FeedbackSessionsService]}
  form="undefined"
  hasCourseLoadingFailed="false"
  hasInstructorsLoadingFailed="false"
  instructorDetailPanels={[Function Array]}
  instructorService={[Function InstructorService]}
  isAddingNewInstructor="false"
  isCourseLoading={[Function Boolean]}
  isEditingCourse="false"
  isInstructorsLoading="false"
  isSavingCourseEdit="false"
  isSavingNewInstructor="false"
  navigationService={[Function NavigationService]}
  newInstructorPanel={[Function Object]}
  ngbModal={[Function NgbModal]}
  originalCourse={[Function Object]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
  timezones={[Function Array]}
>
  <tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    <div
      class="right top-padded"
    >
      <a
        class="small"
        href="/web/instructor/help?questionId=course-instructor-access&section=courses"
        rel="noopener noreferrer"
        target="_blank"
        tmrouterlink="/web/instructor/help"
      >
        <i
          aria-hidden="true"
          class="fa fa-info-circle"
        />
         More about configuring access permissions 
      </a>
    </div>
    
    
  </tm-loading-retry><tm-loading-retry>
    
    
    <div>
      
    </div>
    
    
  </tm-loading-retry><div
    class="text-center"
  >
    <button
      class="btn btn-success"
      id="btn-add-instructor"
      tabindex="6"
      type="button"
    >
       Add New Instructor 
    </button>
  </div>
</tm-instructor-course-edit-page>
`;

exports[`InstructorCourseEditPageComponent should snap with default fields 1`] = `
<tm-instructor-course-edit-page
  CoursesSectionQuestions={[Function Object]}
  EditMode={[Function Object]}
  FormValidator={[Function Object]}
  Sections={[Function Object]}
  allSections={[Function Array]}
  allSessions={[Function Array]}
  authService={[Function AuthService]}
  course={[Function Object]}
  courseId="undefined"
  courseService={[Function CourseService]}
  currInstructorCoursePrivilege={[Function Object]}
  currInstructorGoogleId=""
  feedbackSessionsService={[Function FeedbackSessionsService]}
  form="undefined"
  hasCourseLoadingFailed="false"
  hasInstructorsLoadingFailed="false"
  instructorDetailPanels={[Function Array]}
  instructorService={[Function InstructorService]}
  isAddingNewInstructor="false"
  isCourseLoading={[Function Boolean]}
  isEditingCourse="false"
  isInstructorsLoading="false"
  isSavingCourseEdit="false"
  isSavingNewInstructor="false"
  navigationService={[Function NavigationService]}
  newInstructorPanel={[Function Object]}
  ngbModal={[Function NgbModal]}
  originalCourse={[Function Object]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
  timezones={[Function Array]}
>
  <tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    <div
      class="right top-padded"
    >
      <a
        class="small"
        href="/web/instructor/help?questionId=course-instructor-access&section=courses"
        rel="noopener noreferrer"
        target="_blank"
        tmrouterlink="/web/instructor/help"
      >
        <i
          aria-hidden="true"
          class="fa fa-info-circle"
        />
         More about configuring access permissions 
      </a>
    </div>
    
    
  </tm-loading-retry><tm-loading-retry>
    
    
    <div>
      
    </div>
    
    
  </tm-loading-retry><div
    class="text-center"
  >
    <button
      class="btn btn-success"
      id="btn-add-instructor"
      tabindex="6"
      type="button"
    >
       Add New Instructor 
    </button>
  </div>
</tm-instructor-course-edit-page>
`;

exports[`InstructorCourseEditPageComponent should snap with some instructor details 1`] = `
<tm-instructor-course-edit-page
  CoursesSectionQuestions={[Function Object]}
  EditMode={[Function Object]}
  FormValidator={[Function Object]}
  Sections={[Function Object]}
  allSections={[Function Array]}
  allSessions={[Function Array]}
  authService={[Function AuthService]}
  course={[Function Object]}
  courseId="undefined"
  courseService={[Function CourseService]}
  currInstructorCoursePrivilege={[Function Object]}
  currInstructorGoogleId=""
  feedbackSessionsService={[Function FeedbackSessionsService]}
  form="undefined"
  hasCourseLoadingFailed="false"
  hasInstructorsLoadingFailed="false"
  instructorDetailPanels={[Function Array]}
  instructorService={[Function InstructorService]}
  isAddingNewInstructor="false"
  isCourseLoading={[Function Boolean]}
  isEditingCourse="false"
  isInstructorsLoading="false"
  isSavingCourseEdit="false"
  isSavingNewInstructor="false"
  navigationService={[Function NavigationService]}
  newInstructorPanel={[Function Object]}
  ngbModal={[Function NgbModal]}
  originalCourse={[Function Object]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  timezoneService={[Function TimezoneService]}
  timezones={[Function Array]}
>
  <tm-loading-retry>
    
    
    <tm-loading-spinner>
      <div
        class="loading-container"
      >
        <div
          class="row spinner-border text-primary"
          role="status"
        />
        <div
          class="row text-secondary"
        >
          Loading...
        </div>
      </div>
    </tm-loading-spinner>
    <div
      class="right top-padded"
    >
      <a
        class="small"
        href="/web/instructor/help?questionId=course-instructor-access&section=courses"
        rel="noopener noreferrer"
        target="_blank"
        tmrouterlink="/web/instructor/help"
      >
        <i
          aria-hidden="true"
          class="fa fa-info-circle"
        />
         More about configuring access permissions 
      </a>
    </div>
    
    
  </tm-loading-retry><tm-loading-retry>
    
    
    <div>
      
      
      <tm-instructor-edit-panel>
        <div
          class="card bg-light top-padded"
        >
          <div
            class="card-header bg-primary text-white text-center"
          >
            <strong>
              Instructor 1
            </strong>
            <div
              class="card-header-btn-toolbar"
            >
              
              
              <button
                class="btn btn-primary btn-sm"
                id="btn-edit-instructor-1"
              >
                
                <i
                  aria-hidden="true"
                  class="fa fa-pen"
                />
                 Edit 
              </button>
              
              
              <button
                class="btn btn-primary btn-sm"
                id="btn-delete-instructor-1"
                ngbtooltip="Delete the instructor from the course"
                placement="top-right"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-trash"
                />
                 Delete 
              </button>
              
            </div>
          </div>
          <div
            class="card-body text-center"
          >
            
            <div
              class="form-row form-group"
            >
              <label
                class="col-sm-3 control-label"
              >
                 Google ID: 
              </label>
              
              <div
                class="col-sm-9"
              >
                <input
                  class="form-control ng-untouched ng-pristine ng-valid"
                  disabled=""
                  id="google-id-instructor-1"
                  maxlength="254"
                  tabindex="3"
                />
              </div>
              
            </div>
            <div
              class="form-row form-group"
            >
              <label
                class="col-sm-3 control-label"
              >
                 Name: 
              </label>
              <div
                class="col-sm-9"
              >
                <input
                  class="form-control ng-untouched ng-pristine ng-valid"
                  id="name-instructor-1"
                  maxlength="100"
                  tabindex="4"
                />
              </div>
            </div>
            <div
              class="form-row form-group"
            >
              <label
                class="col-sm-3 control-label"
              >
                 Email: 
              </label>
              <div
                class="col-sm-9"
              >
                <input
                  class="form-control ng-untouched ng-pristine ng-valid"
                  id="email-instructor-1"
                  maxlength="254"
                  tabindex="5"
                />
              </div>
            </div>
            <div
              class="form-row form-group"
            >
              <label
                class="col-sm-3 control-label"
              >
                <input
                  checked=""
                  class="ng-untouched ng-pristine ng-valid"
                  id="checkbox-display-instructor-1"
                  type="checkbox"
                />
                   
                <span
                  class="ngb-tooltip-class"
                  ngbtooltip="If this is unselected, the instructor will be completely invisible to students. E.g. to give access to a colleague for ‘auditing’ your course"
                >
                  Display to students as:
                </span>
              </label>
              <div
                class="col-sm-9"
              >
                <input
                  class="form-control ng-untouched ng-pristine ng-valid"
                  id="displayed-name-instructor-1"
                  placeholder="E.g. Co-lecturer, Teaching Assistant (default: Instructor)"
                  type="text"
                  value="Instructor"
                />
              </div>
            </div>
            <div
              class="form-row form-group align-items-start text-left"
            >
              <label
                class="col-sm-3 control-label text-center text-md-right"
              >
                 Access Level: 
              </label>
              <div
                class="col-sm-9"
                id="access-levels-instructor-1"
              >
                
                
                
                <span
                  id="role-instructor-1"
                >
                  Custom: No access by default. Any access needs to be granted explicitly.
                </span>
                
                <button
                  class="btn btn-link padding-0"
                  type="button"
                >
                  View Details
                </button>
                
              </div>
            </div>
            
            
          </div>
        </div>
      </tm-instructor-edit-panel>
      <br />
    </div>
    
    
  </tm-loading-retry><div
    class="text-center"
  >
    <button
      class="btn btn-success"
      id="btn-add-instructor"
      tabindex="6"
      type="button"
    >
       Add New Instructor 
    </button>
  </div>
</tm-instructor-course-edit-page>
`;
